TicTacToe.java

```diff
@@ -0,0 +1,52 @@
+import java.util.*;
+
+public class TicTacToe {
+
+  public static boolean checkCol(int x_coord, int[][] board) {
+    return (board[0][x_coord] == board[1][x_coord]) && (board[1][x_coord] == board[2][x_coord]);
+  }
+
+  public static boolean checkRow(int y_coord, int[][] board) {
+    return (board[y_coord][0] == board[y_coord][1]) && (board[y_coord][1] == board[y_coord][2]);
+  }
+
+  public static boolean checkRightDiagonal(int x_coord, int y_coord, int[][] board) {
+    if(board[0][0] == board[1][1] && board[1][1] == board[2][2]) return true;
```

Comment:
ordering is wrong here, the x_coord != y_coord logic is a shortcircuit to make sure that we don't do more complicated logic than necessary.

should be something like:
```
if(x_coord != y_coord) return false; // known as guard clauses

return (board[0][0] == board[1][1]) && (board[1][1] == board[2][2]);
```

Reply:
the logic on for the board consists of 6 operations: 4 data fetch from 2D array and 2 comparisons.
vs
the logic for just the coords consists of 2 operations: 2 comparisons

Reply:
also becareful:
(board[0][0] == board[1][1]) && (board[1][1] == board[2][2])
is not the same as
board[0][0] == board[1][1] && board[1][1] == board[2][2]